src/adapters/sources/rentals.ts

```diff
@@ -21,25 +22,29 @@ export function createRentalsNFTSource(
   async function enhanceRentalListing(
     rentals: RentalListing[]
   ): Promise<Sortable<NFTResult, NFTSortBy>[]> {
-    const nftResults: NFTResult[] = await Promise.all(
-      rentals.map(async (rental) => {
-        const nftResult = await nfts.fetchOne(
-          rental.contractAddress,
-          rental.tokenId
-        )
-        if (!nftResult) {
+    const tokenIdsOfRentals = rentals.map((rental) => rental.nftId)
+    const nftResultsOfRentals = await nfts.fetchByTokenIds(tokenIdsOfRentals)
+    const nftResultsOfRentalsById = Object.fromEntries(
+      nftResultsOfRentals.map((nft) => [nft.nft.id, nft])
```

Comment:
just a heads up, this is looping the `nftResultsOfRentals` twice (one for the map and another for the fromEntries) and creating N arrays + 1 object. Using a .reduce would loop only once and create just 1 object.

Reply:
If this is meant just for a single page of 20 elements might not make a difference but for 1000 results it could.

Reply:
Changed to a reduce!